pkg/networkservicemesh/mechanisms/memif/client.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2020 Cisco Systems, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package memif provides networkservice chain elements that support the memif Mechanism
+package memif
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/ligato/vpp-agent/api/models/vpp"
+	vpp_interfaces "github.com/ligato/vpp-agent/api/models/vpp/interfaces"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection/mechanisms/cls"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection/mechanisms/memif"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/networkservice"
+
+	"github.com/networkservicemesh/sdk-vppagent/pkg/networkservicemesh/vppagent"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservicemesh/core/next"
+)
+
+type memifClient struct {
+	baseDir string
+}
+
+// NewClient provides a NetworkServiceClient chain elements that support the memif Mechanism
+func NewClient(baseDir string) networkservice.NetworkServiceClient {
+	return &memifClient{baseDir: baseDir}
+}
+
+func (m *memifClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*connection.Connection, error) {
```

Comment:
opts isn't used here, is this intentional?

Reply:
Fixed